### ISSUE TITLE
Video Call ringing in caller side but not showing incoming call in receiver side

### ISSUE THREAD

**Issue by murugaprabum** [QuickBlox/quickblox-ios-sdk#193]
I am using Quickblox for video chat in iOS. I face a peculiar problem. The problem is as follows.

1. Caller calls the recipient
2. A ringing sound is heard in the caller side
3. There is no Incoming call view controller shown in the recipient side
(Both the phones are in the same network)

Below is the SDP messages that is transferred from the caller. One thing is very peculiar in this. The connection IP address for the audio media and the connection IP address for the video media are different. @QuickBlox-team  Can a call be made by using different relay servers for audio and video. The call works if the same relay server is used for both audio and video.

##### AUDIO MEDIA

```
m=audio 34397 RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 127 126
c=IN IP4 52.20.147.93
```

##### VIDEO MEDIA

```
m=video 49065 RTP/SAVPF 100 116 117 96
c=IN IP4 52.1.106.254
```

##### SDP MESSAGE

```
o=- 254488952483853967 2 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE audio video
a=msid-semantic: WMS ARDAMS
m=audio 34397 RTP/SAVPF 111 103 104 9 102 0 8 106 105 13 127 126
c=IN IP4 52.20.147.93
a=rtcp:58339 IN IP4 52.1.106.254
a=candidate:2087201215 1 udp 2122260223 192.168.1.14 55832 typ host generation 0
a=candidate:2176871851 1 udp 2122194687 10.64.167.130 62613 typ host generation 0
a=candidate:2087201215 2 udp 2122260222 192.168.1.14 59759 typ host generation 0
a=candidate:2176871851 2 udp 2122194686 10.64.167.130 61136 typ host generation 0
a=candidate:854316367 1 tcp 1518280447 192.168.1.14 51894 typ host tcptype passive generation 0
a=candidate:3477311835 1 tcp 1518214911 10.64.167.130 51895 typ host tcptype passive generation 0
a=candidate:854316367 2 tcp 1518280446 192.168.1.14 51896 typ host tcptype passive generation 0
a=candidate:3477311835 2 tcp 1518214910 10.64.167.130 51897 typ host tcptype passive generation 0
a=candidate:2323140715 2 udp 1686052606 203.193.158.234 59759 typ srflx raddr 192.168.1.14 rport 59759 generation 0
a=candidate:2323140715 1 udp 1686052607 203.193.158.234 55832 typ srflx raddr 192.168.1.14 rport 55832 generation 0
a=candidate:3697380798 2 udp 41885438 52.1.106.254 58339 typ relay raddr 203.193.158.234 rport 59759 generation 0
a=candidate:183613341 1 udp 41885439 52.20.147.93 34397 typ relay raddr 203.193.158.234 rport 55832 generation 0
a=candidate:183613341 2 udp 25108478 52.20.147.93 53651 typ relay raddr 203.193.158.234 rport 51890 generation 0
a=candidate:3697380798 1 udp 25108479 52.1.106.254 52618 typ relay raddr 203.193.158.234 rport 51889 generation 0
a=ice-ufrag:zLmAOPKphFPT/vxO
a=ice-pwd:6RUvVGQmUal1wKVrcaWgH0eX
a=fingerprint:sha-256 F7:72:E1:18:4A:3B:DF:36:EF:FE:DC:3D:81:98:69:85:0D:DF:89:08:0B:07:FC:80:A4:00:B9:D7:31:4F:B7:13
a=setup:actpass
a=mid:audio
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=sendrecv
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=fmtp:111 minptime=10; useinbandfec=1
a=rtpmap:103 ISAC/16000
a=rtpmap:104 ISAC/32000
a=rtpmap:9 G722/8000
a=rtpmap:102 ILBC/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:106 CN/32000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:127 red/8000
a=rtpmap:126 telephone-event/8000
a=maxptime:60
a=ssrc:911692275 cname:V6s86EwWz6S/qhV7
a=ssrc:911692275 msid:ARDAMS ARDAMSa0
a=ssrc:911692275 mslabel:ARDAMS
a=ssrc:911692275 label:ARDAMSa0
m=video 49065 RTP/SAVPF 100 116 117 96
c=IN IP4 52.1.106.254
a=rtcp:49078 IN IP4 52.20.147.93
a=candidate:2087201215 1 udp 2122260223 192.168.1.14 52135 typ host generation 0
a=candidate:2176871851 1 udp 2122194687 10.64.167.130 55998 typ host generation 0
a=candidate:2087201215 2 udp 2122260222 192.168.1.14 57765 typ host generation 0
a=candidate:2176871851 2 udp 2122194686 10.64.167.130 61925 typ host generation 0
a=candidate:854316367 1 tcp 1518280447 192.168.1.14 51898 typ host tcptype passive generation 0
a=candidate:3477311835 1 tcp 1518214911 10.64.167.130 51899 typ host tcptype passive generation 0
a=candidate:854316367 2 tcp 1518280446 192.168.1.14 51900 typ host tcptype passive generation 0
a=candidate:3477311835 2 tcp 1518214910 10.64.167.130 51901 typ host tcptype passive generation 0
a=candidate:2323140715 1 udp 1686052607 203.193.158.234 52135 typ srflx raddr 192.168.1.14 rport 52135 generation 0
a=candidate:2323140715 2 udp 1686052606 203.193.158.234 57765 typ srflx raddr 192.168.1.14 rport 57765 generation 0
a=candidate:3697380798 1 udp 41885439 52.1.106.254 49065 typ relay raddr 203.193.158.234 rport 52135 generation 0
a=candidate:183613341 2 udp 41885438 52.20.147.93 49078 typ relay raddr 203.193.158.234 rport 57765 generation 0
a=candidate:183613341 1 udp 25108479 52.20.147.93 64080 typ relay raddr 203.193.158.234 rport 51886 generation 0
a=candidate:3697380798 2 udp 25108478 52.1.106.254 37394 typ relay raddr 203.193.158.234 rport 51891 generation 0
a=ice-ufrag:zLmAOPKphFPT/vxO
a=ice-pwd:6RUvVGQmUal1wKVrcaWgH0eX
a=fingerprint:sha-256 F7:72:E1:18:4A:3B:DF:36:EF:FE:DC:3D:81:98:69:85:0D:DF:89:08:0B:07:FC:80:A4:00:B9:D7:31:4F:B7:13
a=setup:actpass
a=mid:video
a=extmap:2 urn:ietf:params:rtp-hdrext:toffset
a=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:4 urn:3gpp:video-orientation
a=sendrecv
a=rtcp-mux
a=rtpmap:100 VP8/90000
a=rtcp-fb:100 ccm fir
a=rtcp-fb:100 nack
a=rtcp-fb:100 nack pli
a=rtcp-fb:100 goog-remb
a=rtpmap:116 red/90000
a=rtpmap:117 ulpfec/90000
a=rtpmap:96 rtx/90000
a=fmtp:96 apt=100
a=ssrc-group:FID 409187601 2299817338
a=ssrc:409187601 cname:V6s86EwWz6S/qhV7
a=ssrc:409187601 msid:ARDAMS ARDAMSv0
a=ssrc:409187601 mslabel:ARDAMS
a=ssrc:409187601 label:ARDAMSv0
a=ssrc:2299817338 cname:V6s86EwWz6S/qhV7
a=ssrc:2299817338 msid:ARDAMS ARDAMSv0
a=ssrc:2299817338 mslabel:ARDAMS
a=ssrc:2299817338 label:ARDAMSv0
```

**Comment by Pro100Andrey** [comment]
Hi @murugaprabum ! Please check new version 2.0 https://github.com/QuickBlox/quickblox-ios-sdk/tree/master/sample-videochat-webrtc

**Comment by murugaprabum** [comment]
@Pro100Andrey How is this problem fixed in the new version? Can you give us more details about it?

**Comment by murugaprabum** [comment]
@QuickBlox-team Can you let us know how is this issue fixed in the latest version of quickblox? We are evaluating your product for use in one of our apps. Would be great if you could share some info on how this problem has been fixed. How can I convince my higher authorities that this problem has been fixed in the latest version.

**Comment by iSevenDays** [comment]
@murugaprabum Hi! Please take look at the link below
http://www.quickblox.com/developers/Sample-webrtc-ios

There you can find more information.

Also there are many problems has been fixed since version 1.0.6.

At this time we use the latest WebRTC revision, so all problems since v1.0.6 that are related to webrtc issues are solved.

Answer to your first question: yes, it is possible. TURN server must have at least two IP, so if there is connection through our TURN server, yes, you can have different IP.

>> There is no Incoming call view controller shown in the recipient side (Both the phones are in the same network)

Please run our sample video chat and provide us with full log. [QBSettings setLogLevel: QBLogLevelDebug];
